misk-datadog/src/main/kotlin/misk/tracing/backends/datadog/DatadogTracingBackendModule.kt

```diff
@@ -8,16 +8,28 @@ import misk.inject.KAbstractModule
 /**
  * Binds the datadog tracer to opentracing's [Tracer]
  */
-class DatadogTracingBackendModule : KAbstractModule() {
+
+class DatadogTracingBackendModule() : KAbstractModule() {
+
+  private var useAutoMdcInjection: Boolean = false
+
+  @Deprecated("useAutoMdcInjection is an experimental option that will be removed in a future version")
+  constructor(useAutoMdcInjection: Boolean): this() {
+    this.useAutoMdcInjection = useAutoMdcInjection
+  }
```

Comment:
I have added this as a seperate constructor instead of a defaulted value in the primary constructor to reduce the changes to the API as we have seen examples recently where default arguments can actually result in runtime errors when version conflicts are resolved in consumers. That said, with the default argument, the API changes look like:

```diff
public final class misk/tracing/backends/datadog/DatadogTracingBackendModule : misk/inject/KAbstractModule {
	public fun <init> ()V
+	public fun <init> (Z)V
+	public synthetic fun <init> (ZILkotlin/jvm/internal/DefaultConstructorMarker;)V
}
```

I think this would still be backwards compatible for existing consumer, however I'm not confident about it and given this will hopefully be a short lived option, I'm ok with it being a bit messier in the implementation for the added piece of mind.  However, if others strongly prefer the default value I can change it over before merging.